mypy/messages.py

```diff
@@ -819,6 +819,21 @@ def incompatible_typevar_value(self, callee: CallableType, index: int,
         self.fail('Type argument {} of {} has incompatible value {}'.format(
             index, callable_name(callee), self.format(type)), context)
 
+    def incompatible_constrained_arguments(self,
+                                           callee: CallableType,
+                                           index: int,
+                                           constraints: Mapping[str, Sequence[str]],
+                                           context: Context) -> None:
+        for key, values in constraints.items():
+            self.fail('Type argument {} of {} has incompatible value'.format(
+                index, callable_name(callee)), context)
+            if len(values) == 2:
+                constraint_str = '{} or {}'.format(values[0], values[1])
+            elif len(values) > 3:
+                constraint_str = ', '.join(values[:-1]) + ', or ' + values[-1]
+            self.note('"{}" must be all one type: {}'.format(
```

Comment:
I think I can probably find the argument indices easily enough, but the types of the other arguments isn't available when constructing the message because the message builder only gets called when the second argument is analyzed (from what I can tell, please let me know if there is something I have overlooked). I can get `mypy.nodes.StrExpr` and `mypy.nodes.BytesExpr` from `context`, but I do not know how to safely get these values as `MessageBuilder.format` for both of these returns `object`.